src/test/java/brave/features/opentracing/OpenTracingAdapterTest.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.features.opentracing;
+
+import brave.Tracer;
+import brave.propagation.TraceContext;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapExtractAdapter;
+import io.opentracing.propagation.TextMapInjectAdapter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import zipkin.Constants;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.assertj.core.data.MapEntry.entry;
+import static zipkin.internal.Util.UTF_8;
+
+/**
+ * This shows how one might make an OpenTracing adapter for Brave, and how to navigate in and out
+ * of the core concepts.
+ */
+public class OpenTracingAdapterTest {
+    List<zipkin.Span> spans = new ArrayList<>();
+    Tracer brave = Tracer.newBuilder().reporter(spans::add).build();
+    BraveTracer opentracing = BraveTracer.wrap(brave);
+
+    @Test public void startWithOpenTracingAndFinishWithBrave() {
+        io.opentracing.Span openTracingSpan = opentracing.buildSpan("encode")
+                .withTag(Constants.LOCAL_COMPONENT, "codec")
+                .withStartTimestamp(1L).start();
+
+        brave.Span braveSpan = ((BraveSpan) openTracingSpan).unwrap();
+
+        braveSpan.annotate(2L, "pump fake");
+        braveSpan.finish(3L);
+
+        checkSpanReportedToZipkin();
+    }
+
+    @Test public void startWithBraveAndFinishWithOpenTracing() {
+        brave.Span braveSpan = brave.newTrace().name("encode")
+                .tag(Constants.LOCAL_COMPONENT, "codec")
+                .start(1L);
+
+        io.opentracing.Span openTracingSpan = BraveSpan.wrap(braveSpan);
+
+        openTracingSpan.log(2L, "pump fake");
+        openTracingSpan.finish(3L);
+
+        checkSpanReportedToZipkin();
+    }
+
+    @Test
+    public void extractTraceContext() throws Exception {
```

Comment:
I had a look at this, and it is an interesting topic.

Seems the api says to throw IllegalArgumentException when you cannot extract a context due to something corrupt https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/Tracer.java#L76

It doesn't say if null is permitted or not as a return value. I just noticed that at least in Jaeger null is returned when fields are missing etc.

sampled=0 without ids isn't corrupt, rather an unsampled context. Probably grey area whether to generate span ids here when absent (Historically in B3, ids were ignored when unsampled, but some prefer to propagate them anyway)